Crash while depositing money

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues for each one!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong

Ivan has $50 in his inventory. If I deposit that money, the game will instantly crash to desktop.

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
| v0.22.0-20240722+f7de10b | JA2 1.12 GOG |    no    |    yes     |

[//]: # (Please also include:)
[//]: # (- how to reproduce the bug and, if possible, a save game)
[//]: # (- the operating system)
[//]: # (- in case you know, for enhancements, if this feature was already added by some other source mod like 1.13)

Microsoft Windows 11 Pro 10.0.22631 Build 22631

If you go into tactical mode and move the $50 to the Deposit icon, then deposit the money, the game will crash to desktop. The error in the log is:
``` [ERROR] sgp\SGP.cc: Game has been terminated due to an unrecoverable error: failed to get dealer: invalid dealer ID: -1 ```
A recovery savegame is automatically created (error.sav), but loading the save game will not load to the strategic map, it will just play the background music and hang on the loading image once the game has finished loading.

[//]: # (When pasting backtraces or any other logs with hashed enumerations, please format it as a code block.)
[//]: # (You can do that by selecting it and clicking the "<>" icon or manually putting three backticks [```] at each end.)

[depositing-money_Error.zip](https://github.com/user-attachments/files/16335974/depositing-money_Error.zip)
[ja2.zip](https://github.com/user-attachments/files/16335975/ja2.zip)
[confs.zip](https://github.com/user-attachments/files/16335976/confs.zip)

--- a/src/game/Tactical/Arms_Dealer_Init.cc
+++ b/src/game/Tactical/Arms_Dealer_Init.cc
@@ -887,7 +887,7 @@ ArmsDealerID GetArmsDealerIDFromMercID(UINT8 const ubMercID)
 
 ArmsDealerType GetTypeOfArmsDealer(ArmsDealerID ubDealerID)
 {
-	if (ubDealerID >= NUM_ARMS_DEALERS)
+	if (ubDealerID < 0 || ubDealerID >= NUM_ARMS_DEALERS)
 	{
 		return ArmsDealerType::NOT_VALID_DEALER;
 	}

